### ISSUE TITLE
Fails when phantomjs pkg.path has a space

### ISSUE THREAD

**Issue by ts-web** [macbre/phantomas#505]
When I run the command on my Windows box, I get this error:

    'C:\Program' is not recognized as an internal or external command, operable program or batch file.

I traced it down to **lib/engines.js**:109

		proc = spawn(engine.path, engineArgs, {
			env: process.env
		});

`engine.path` was set to `C:\Program Files\nodejs\phantomjs.CMD` *(Note the space).* This was causing the error.

engine.path comes from **lib/engines/webkit.js**

    /**
     * Defines PhantomJS engine
     */
    var pkg = require('phantomjs');
    
    module.exports = {
    	name: 'PhantomJS',
    	engine: 'webkit',
    	path: pkg.path,
    	version: pkg.version,
    	getUserAgent: function() {
    		return 'PhantomJS/' + pkg.version
    	}
    };

Line of interest: `path: pkg.path,`

And I trace that back to **\node_modules\phantomjs\lib\location.js**

    module.exports.location = "C:\\Program Files\\nodejs\\phantomjs.CMD"

Can anyone shed some light on this problem?

**Comment by macbre** [comment]
It seems that spaces need to be escaped. I'll take a look and ask you for assistance in testing the fix (I do not have access to Windows machine).

**Comment by ts-web** [comment]
Also, this only happens when I use phantomas in JavaScript. (It works from the command line when installed globally.) Here is how I am using it in JS:

```javascript
var phantomas = require('phantomas');

phantomas('http://example.com', {"analyze-css": true}, function(err, json, results) {
  console.log([
    'phantomas results',
    err, // null or exit code from phantomas process
    json, // parsed JSON with raw results
    results // results object with metrics values, offenders, asserts data
  ]);
});
```

P.S. Yes sure, I'd be willing to assist.

**Comment by ts-web** [comment]
I am not experiencing the same problem anymore. I can't think of what fixed it besides what I did in my previous comment. I must have forgotten to re-check the JS code I cited; I just assumed that it was still broken. Silly me.

I think this problem was caused by a malfunction during the process of my initial installation. That is why an uninstall and reinstall fixed it. The malfunction was related to [this bug](https://github.com/macbre/analyze-css/issues/89), which popped up while installing the dependency `analyze-css`. It caused my RAM to be used up and my computer to freeze, forcing me to restart. This caused filesystem corruptions that I had to fix via chkdsk, so it isn't to far-fetched to think that it corrupted the global phantomas installation as well.

So I think this bug can be closed.
